Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp

```diff
@@ -142,6 +142,7 @@ void WebFrameLoaderClient::dispatchDecidePolicyForNavigationAction(const Navigat
         coreFrame ? coreFrame->loader().effectiveSandboxFlags() : SandboxFlags(),
         navigationAction.privateClickMeasurement(),
         requestingFrame ? requestingFrame->advancedPrivacyProtections() : OptionSet<AdvancedPrivacyProtections> { },
+        requestingFrame ? requestingFrame->originatorAdvancedPrivacyProtections() : OptionSet<AdvancedPrivacyProtections> { },
```

Comment:
Do (or will) we need this information for anything else in the future? Could you just do something like, instead:
```
requestingFrame ? (redirectResponse.isNull() ? requestingFrame->advancedPrivacyProtections() : requestingFrame->originatorAdvancedPrivacyProtections()) : OptionSet<AdvancedPrivacyProtections> { },
```

This seems like it would avoid some of the complexity.

Reply:
I realize this wasn't exactly clear. I mean instead of adding the new entry, just set the `advancedPrivacyProtections` entry with the correct value (if this is a correct assumption).

Reply:
I don't think that will work, since we want to cover both client and server side redirection. Checking the response to see if there was a redirect will only cover server side redirection.

The tests which use `window.location = x` cover this.

Reply:
Okay, that makes sense. There's probably another way to do this from WebCore based on FrameLoader::changeLocation, but your patch looks fine.